api/unstable/crl.h

```diff
@@ -0,0 +1,181 @@
+/*
+* Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+*
+* Licensed under the Apache License, Version 2.0 (the "License").
+* You may not use this file except in compliance with the License.
+* A copy of the License is located at
+*
+*  http://aws.amazon.com/apache2.0
+*
+* or in the "license" file accompanying this file. This file is distributed
+* on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+* express or implied. See the License for the specific language governing
+* permissions and limitations under the License.
+*/
+
+#pragma once
+
+#include <s2n.h>
+
+/**
+ * @file crl.h
+ *
+ * The following APIs enable applications to determine if a received certificate has been revoked by its CA, via
+ * Certificate Revocation Lists (CRLs). Please see the CRL Validation section in the usage guide for more information.
+ *
+ * The CRL APIs are currently considered unstable, since they have been recently added to s2n-tls. After gaining more
+ * confidence in the correctness and usability of these APIs, they will be made stable.
+ *
+ */
+
+struct s2n_crl_lookup;
+
+/**
+ * A callback which can be implemented to provide s2n-tls with CRLs to use for CRL validation.
+ *
+ * This callback is triggered once for each certificate received during the handshake. To provide s2n-tls with a CRL for
+ * the certificate, use `s2n_crl_lookup_set()`. To ignore the certificate and not provide a CRL, use
+ * `s2n_crl_lookup_ignore()`.
+ *
+ * This callback can be synchronous or asynchronous. For asynchronous behavior, return success without calling
+ * `s2n_crl_lookup_set()` or `s2n_crl_lookup_ignore()`. The connection will block until one of these functions is
+ * called.
```

Comment:
Nit: I wonder if we should phrase this as "The connection / s2n_negotiate will return S2N_BLOCKED_ON_APPLICATION_INPUT". I know we generally refer to that as the connection blocking, but it's not traditionally what "blocking" means ;)

Reply:
That makes sense. Added!